### ISSUE TITLE
fastlane 2.85.0 xcodeproj 1.5.6 crashes

### ISSUE THREAD

**Issue by githhhh** [fastlane/fastlane#12086]
```
+------------------+-------------+
|          Lane Context          |
+------------------+-------------+
| DEFAULT_PLATFORM | ios         |
| PLATFORM_NAME    | ios         |
| LANE_NAME        | ios tf_beta |
| FL_CHANGELOG     |             |
+------------------+-------------+

+------+------------------------------------+-------------+
|                    fastlane summary                     |
+------+------------------------------------+-------------+
| Step | Action                             | Time (in s) |
+------+------------------------------------+-------------+
| 1    | default_platform                   | 0           |
| 2    | Switch to ios prepare lane         | 0           |
| 3    | Switch to ios getchangelog lane    | 0           |
| 4    | Switch to ios update_lastcinumber  | 0           |
|      | lane                               |             |
| 5    | changelog_from_git_commits         | 0           |
| 6    | Switch to ios build lane           | 0           |
| 7    | gym                                | 4           |
+------+------------------------------------+-------------+

[18:37:46]: fastlane finished with errors
/Users/bin/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/set.rb:84:in `new': stack level too deep (SystemStackError)
	from /Users/bin/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/set.rb:84:in `initialize'
	from /Users/bin/.rvm/gems/ruby-2.4.1/gems/xcodeproj-1.5.6/lib/xcodeproj/config.rb:54:in `new'
	from /Users/bin/.rvm/gems/ruby-2.4.1/gems/xcodeproj-1.5.6/lib/xcodeproj/config.rb:54:in `block in initialize'
	from /Users/bin/.rvm/gems/ruby-2.4.1/gems/xcodeproj-1.5.6/lib/xcodeproj/config.rb:53:in `each'
	from /Users/bin/.rvm/gems/ruby-2.4.1/gems/xcodeproj-1.5.6/lib/xcodeproj/config.rb:53:in `initialize'
	from /Users/bin/.rvm/gems/ruby-2.4.1/gems/xcodeproj-1.5.6/lib/xcodeproj/config.rb:140:in `new'
	from /Users/bin/.rvm/gems/ruby-2.4.1/gems/xcodeproj-1.5.6/lib/xcodeproj/config.rb:140:in `block in to_hash'
	from /Users/bin/.rvm/gems/ruby-2.4.1/gems/xcodeproj-1.5.6/lib/xcodeproj/config.rb:137:in `map'
	 ... 10905 levels...
	from /Users/bin/.rvm/gems/ruby-2.4.1/bin/fastlane:23:in `load'
	from /Users/bin/.rvm/gems/ruby-2.4.1/bin/fastlane:23:in `<main>'
	from /Users/bin/.rvm/gems/ruby-2.4.1/bin/ruby_executable_hooks:15:in `eval'
	from /Users/bin/.rvm/gems/ruby-2.4.1/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by joshdholtz** [comment]
@githhhh Hey! Are you able to run this with `--verbose` and send the whole output as well as `fastlane env` output?

Also, does your current setup work with if you downgrade `fastlane` or `xcodeproj`?

**Comment by githhhh** [comment]
i find: and fix it 
https://github.com/CocoaPods/Xcodeproj/issues/505

**Comment by elitalon** [comment]
I can confirm this behaviour occurs with fastlane 2.86.0 and xcodeproj 1.5.6.

I'm having problems using `get_version_number`, which internally asks for the "Info.plist" path. We have that path set to `$(SRCROOT)/<redacted>/Supporting Files/Info.plist` and xcodeproj seems to fail to resolve the value of `$(SRCROOT)`.

**Comment by joshdholtz** [comment]
@elitalon Sorry about that! I have a PR out to fix your issue - https://github.com/fastlane/fastlane/pull/12138

**Comment by elitalon** [comment]
@joshdholtz Perfect, thanks!

**Comment by joshdholtz** [comment]
@elitalon The fix for you issue was just pushed in  `2.87.0` 

@githhhh Does `2.87.0` fix your issue? We bumped to `xcodeproj` version `1.5.7`

**Comment by githhhh** [comment]
@joshdholtz Perfect, thanks!
